analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -126,7 +126,6 @@ public override void VisitInvocationExpression(InvocationExpressionSyntax node)
                     }
                     else
                     {
-                        invalidInvocations ??= new();
```

Comment:
This was meant to be initialized only if needed. Please revert.

Reply:
It is meant to avoid null checks in the code. I think it's a good tradeoff. Do we create that many instances?

Reply:
In 99% of the cases we don't have an "invalidInvocation". "invalid" is defined as calling a wrong "Ilogger.Log.." method. Most of the time, we don't have any call to ILogger (no secondary location) or the right Log.. method is called (no diagnostic).